base/logger_file.go

```diff
@@ -352,7 +359,7 @@ func runLogDeletion(ctx context.Context, logDirectory string, logLevel string, s
 	willDelete := false
 	for i := len(files) - 1; i >= 0; i-- {
 		file := files[i]
-		if strings.HasPrefix(file.Name(), logFilePrefix+logLevel) && strings.HasSuffix(file.Name(), ".log.gz") {
+		if strings.HasPrefix(file.Name(), logPrefix) && strings.HasSuffix(file.Name(), ".gz") {
```

Comment:
I basically agree with this - but I thought of a bunch of edge cases which I added tests for. This become really hard to test for without regexp, tests are `TestConsoleIrregularLogPaths`.

Maybe this is just me, but I worry about naming something `log` for temporary debugging, and then in the old way this code worked, anything starting with `log` in the directory would get deleted, which wouldn't necessarily be the way you'd want it. Now it should delete things:

example: `/path/to/foo.log`

1. foo.log
2. foo-2016-11-04T18-30-00.000.log
3. foo-2016-11-04T18-30-00.000.log.gz

The other edge case if you name the file: `.consoleLog` with a leading `.` and no other extension. This was what tipped the point to using a regex since you could no longer rely on a prefix match since there was none for rotated logs.